pkg/sql/sem/tree/function_definition.go

```diff
@@ -28,6 +28,29 @@ type FunctionDefinition struct {
 	FunctionProperties
 }
 
+// Volatility indicates whether the result of a function is dependent *only*
+// on the values of its explicit arguments, or can change due to outside factors
+// (such as parameter variables or table contents).
+//
+// This matches the postgres definition of volatility.
+//
+// NOTE: functions having side-effects, such as setval(),
+// must be labeled volatile to ensure they will not get optimized away,
+// even if the actual return value is not changeable.
+type Volatility byte
+
+const (
+	// VolatilityImmutable indicates the builtin result never changes for a
+	// given input.
+	VolatilityImmutable Volatility = 'i'
+	// VolatilityStable indicates the builtin result never changes during
+	// a given scan.
```

Comment:
The "given scan" part confuses me. I was expecting something like:
```
Immutable: never changes for a given input, even across multiple executions of a statement
Stable: never changes for a given input when invoked multiple times in a single statement, but may change across multiple executions of a statement
Volatile: may change when invoked multiple times in a single statement
```
Examples of each case would be helpful as well, to give people a quick intuition:
```
Immutable: length(col)
Stable: now()
Volatile: random()
```

Reply:
I'll fix this up tomorrow! I have 1000 functions to audit :D